### PR TITLE
Prepare for LitElement & TypeScript codebase conversion

### PR DESCRIPTION
**Pull request by web-padawan**
The PR is ready for review, except the demos that need to be implemented separately.

### CODE REVIEW

File: test/unit/sample.test.ts [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+    expect(element instanceof VaadinComponent).to.be.ok;
+  });
+
+  it('should have a valid static "is" getter', () => {
+    expect(customElements.get(tagName).is).to.equal(tagName);
+  });
+
+  it('should have a valid version number', () => {
+    expect(customElements.get(tagName).version).to.match(/^(\d+\.)?(\d+\.)?(\d+)(-(alpha|beta)\d+)?$/);
+  });
```
Comment (Haprog) [inline test/unit/sample.test.ts]: These should probably be wrapped in a `describe('custom element definition', () => {` block same as in [accordion PR](https://github.com/vaadin/vaadin-accordion/pull/30/files#diff-db5b90a4a9e659e4115ef9fe65ad6e86R37-R38) and [details](https://github.com/vaadin/vaadin-details/blob/next/test/unit/details.test.ts#L23-L24) to make it easier to stay consistent when creating new components based on this.

File: vaadin-component.ts [hunk 2]
```diff
@@ -0,0 +1 @@
+import './theme/lumo/vaadin-component';
```
Comment (Haprog) [inline vaadin-component.ts]: This should probably also have:
```ts
export { VaadinComponent } from './src/vaadin-component';
```
?

File: .travis.yml [hunk 3]
```diff
@@ -1,67 +1,21 @@
...
-      fi;
-    fi
+
+- if [[ "$TRAVIS_EVENT_TYPE" != "pull_request" ]]; then
+    npm run test:sauce && npm run test:visual;
+  else
+    npm run check:version &&
+    npm run lint &&
+    npm test;
+  fi
```
[the remark this answers is not in the document]
Comment (web-padawan) [reply .travis.yml]: Actually, vice versa: we agreed with Tomi to remove `yarn.lock` so I will update accordion and details to use npm.

File: README.md [hunk 4]
```diff
@@ -80,43 +47,38 @@ To use the Material theme, import the correspondent file from the `theme/materia
...
+  `theme/material/vaadin-component.js`
 
-- Alias for `theme/lumo/vaadin-element.html`:
+- Alias for `theme/lumo/vaadin-component.js`:
 
-  `vaadin-element.html`
+  `vaadin-component.js`
 
 
 ## Running demos and tests in a browser
```
[the remark this answers is not in the document]
Comment (web-padawan) [reply README.md]: Nice catch, thanks. Will fix.

File: babel.config.js [hunk 5]
```diff
@@ -0,0 +1,12 @@
...
+  plugins: [
+    [
+      '@babel/plugin-proposal-decorators',
+      {
+        decoratorsBeforeExport: true
+      }
+    ],
+    '@babel/plugin-proposal-class-properties'
+  ],
+}
```
[the remark this answers is not in the document]
Comment (web-padawan) [reply babel.config.js]: Thanks, will do.

File: test/unit/sample.test.ts [hunk 6]
```diff
@@ -0,0 +1,31 @@
+import { fixture, html } from '@vaadin/vaadin-component-dev-dependencies/testing.js';
+import { VaadinComponent } from '../../src/vaadin-component';
+
+const { expect } = chai;
```
[the remark this answers is not in the document]
Comment (web-padawan) [reply test/unit/sample.test.ts]: Thanks, will do.

File: babel.config.js (hunk 5 above)
[the remark this answers is not in the document]
Comment (web-padawan) [reply babel.config.js]: Done.

File: test/unit/sample.test.ts (hunk 6 above)
[the remark this answers is not in the document]
Comment (web-padawan) [reply test/unit/sample.test.ts]: Done.

File: test/unit/sample.test.ts [hunk 7]
```diff
@@ -0,0 +1,31 @@
...
+describe('sample test', () => {
+  let element: VaadinComponent;
+  let tagName: string;
+
+  beforeEach(async () => {
+    element = await fixture(
+      html`
+        <vaadin-component></vaadin-component>
+      `
+    );
```
[the remark this answers is not in the document]
Comment (web-padawan) [reply test/unit/sample.test.ts]: Done.

File: test/unit/sample.test.ts (hunk 1 above)
Comment (web-padawan) [reply test/unit/sample.test.ts]: Done.

File: vaadin-component.ts (hunk 2 above)
Comment (web-padawan) [reply vaadin-component.ts]: Done.

**Pull request merged**